Fix warnings when compiling fsser with clang

fsser.c:152:6: warning: variable 'readfd' is used uninitialized whenever 'if' condition is false
      [-Wsometimes-uninitialized]
        if (device != NULL) {
            ^~~~~~~~~~~~~~
fsser.c:184:21: note: uninitialized use occurs here
        int res = cmd_loop(readfd, writefd);
                           ^~~~~~
fsser.c:152:2: note: remove the 'if' if its condition is always true
        if (device != NULL) {
        ^~~~~~~~~~~~~~~~~~~~
fsser.c:80:31: note: initialize the variable 'readfd' to silence this warning
        serial_port_t writefd, readfd;
                                     ^
                                      = 0
fsser.c:152:6: warning: variable 'writefd' is used uninitialized whenever 'if' condition is false
      [-Wsometimes-uninitialized]
        if (device != NULL) {
            ^~~~~~~~~~~~~~
fsser.c:184:29: note: uninitialized use occurs here
        int res = cmd_loop(readfd, writefd);
                                   ^~~~~~~
fsser.c:152:2: note: remove the 'if' if its condition is always true
        if (device != NULL) {
        ^~~~~~~~~~~~~~~~~~~~
fsser.c:80:23: note: initialize the variable 'writefd' to silence this warning
        serial_port_t writefd, readfd;
                             ^
                              = 0
fsser.c:139:13: warning: variable 'dir' is used uninitialized whenever 'if' condition is true
      [-Wsometimes-uninitialized]
        } else if (argc > i+1) {
                   ^~~~~~~~~~
fsser.c:144:23: note: uninitialized use occurs here
        log_info("dir=%s\n", dir);
                             ^~~
fsser.c:139:9: note: remove the 'if' if its condition is always false
        } else if (argc > i+1) {
               ^~~~~~~~~~~~~~~~~
fsser.c:136:13: warning: variable 'dir' is used uninitialized whenever 'if' condition is true
      [-Wsometimes-uninitialized]
        } else if (i == argc) {
                   ^~~~~~~~~
fsser.c:144:23: note: uninitialized use occurs here
        log_info("dir=%s\n", dir);
                             ^~~
fsser.c:136:9: note: remove the 'if' if its condition is always false
        } else if (i == argc) {
               ^~~~~~~~~~~~~~~~
fsser.c:83:11: note: initialize the variable 'dir' to silence this warning
        char _dir;
                 ^
                  = NULL
4 warnings generated.
fscmd.c:166:9: warning: variable 'rv' is used uninitialized whenever 'if' condition is true
      [-Wsometimes-uninitialized]
                                if ((p - argv[i] - 4) > MAX_LEN_OF_PROVIDER_NAME) {
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
fscmd.c:181:8: note: uninitialized use occurs here
                        if (rv < 0) {
                            ^~
fscmd.c:166:5: note: remove the 'if' if its condition is always false
                                if ((p - argv[i] - 4) > MAX_LEN_OF_PROVIDER_NAME) {
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
fscmd.c:158:10: note: initialize the variable 'rv' to silence this warning
                        int rv;
                              ^
                               = 0
1 warning generated.
fs_provider.c:678:5: warning: variable 'file' is used uninitialized whenever 'if' condition is false
      [-Wsometimes-uninitialized]
        if(fp) {
           ^~
fs_provider.c:716:93: note: uninitialized use occurs here
  ...%s (@ %p))=%p (fp=%p)\n", options, filename, filename, (void_)file, (void_)fp);
                                                                   ^~~~
fs_provider.c:678:2: note: remove the 'if' if its condition is always true
        if(fp) {
        ^~~~~~~
fs_provider.c:570:12: note: initialize the variable 'file' to silence this warning
        File *file;
                  ^
                   = NULL
1 warning generated.
tcp_provider.c:325:13: warning: variable 'file' is used uninitialized whenever 'if' condition is
      true [-Wsometimes-uninitialized]
        if (ap == NULL) {               /_ No address succeeded _/
            ^~~~~~~~~~
tcp_provider.c:343:85: note: uninitialized use occurs here
  ...%s:%s =%p (fd=%d)\n",mode, tnep->hostname, buf, (void_)file, sockfd);
                                                            ^~~~
tcp_provider.c:325:9: note: remove the 'if' if its condition is always false
        if (ap == NULL) {               /\* No address succeeded */
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tcp_provider.c:266:12: note: initialize the variable 'file' to silence this warning
        File *file;
                  ^
                   = NULL
1 warning generated.

--- a/pcserver/fs_provider.c
+++ b/pcserver/fs_provider.c
@@ -567,7 +567,7 @@ static void close_fds(endpoint_t *ep, int tfd) {
 // open a file for reading, writing, or appending
 static int open_file(endpoint_t *ep, int tfd, const char *buf, const char *opts, int *reclen, int fs_cmd) {
 	int er = CBM_ERROR_FAULT;
-	File *file;
+	File *file=NULL;
 
 	uint16_t recordlen = 0;
 	uint8_t type;

--- a/pcserver/fscmd.c
+++ b/pcserver/fscmd.c
@@ -155,7 +155,7 @@ void cmd_assign_from_cmdline(int argc, char *argv[]) {
 			}
 	
 			// int rv = provider_assign(argv[i][2] & 0x0f, &(argv[i][4]));
-			int rv;
+			int rv=0;
 			int drive = argv[i][2] & 0x0f;
 			char provider_name[MAX_LEN_OF_PROVIDER_NAME + 1];
 			char *provider_parameter;

--- a/pcserver/fsser.c
+++ b/pcserver/fsser.c
@@ -77,10 +77,10 @@ void usage(int rv) {
 
 
 int main(int argc, char *argv[]) {
-	serial_port_t writefd, readfd;
+	serial_port_t writefd=0, readfd=0;
 	serial_port_t fdesc;
 	int i;
-	char *dir;
+	char *dir=NULL;
 	char *device = NULL;	/* device name or NULL if stdin/out */
 	char parameter_d_given = FALSE;
 

--- a/pcserver/tcp_provider.c
+++ b/pcserver/tcp_provider.c
@@ -263,7 +263,7 @@ static void close_fds(endpoint_t *ep, int tfd) {
 static int open_file(endpoint_t *ep, int tfd, const char *buf, const char *mode) {
 	int ern;
 	int er = CBM_ERROR_FAULT;
-	File *file;
+	File *file=NULL;
 	struct addrinfo *addr, *ap;
 	struct addrinfo hints;
 	int sockfd;

